src/ActionHistory.js

```diff
@@ -0,0 +1,38 @@
+import React, { PropTypes } from 'react';
+import Action from './Action';
+
+const ActionHistory = (props) => {
+  return (
+    <div className="side-bar action-history">
+      <h3>Action History</h3>
+      {props.actions.length > 0 ? null : 'Perform an action, jagweed.'}
+      <div className="side-bar-scroll">
+        <ul className="action-list">
+          {props.actions.slice(0, 10).map(action => {
```

Comment:
Instead of keeping all history items in memory, you could also implement the limit when adding items to the queue. If you have more than 10 items right now and then remove some, the old items reappear. Since we have the expiration time, it's not that big of a deal, otherwise you would have a memory leak in your application and that action history could grow endlessly.